mono/metadata/handle.h

```diff
@@ -401,19 +401,15 @@ This is why we evaluate index and value before any call to MONO_HANDLE_RAW or ot
 #define MONO_HANDLE_ARRAY_SETREF(HANDLE, IDX, VALUE) do {	\
 		int __idx = (IDX);	\
    		MonoObjectHandle __val = MONO_HANDLE_CAST (MonoObject, VALUE);		\
-		do {							\
-			MONO_HANDLE_SUPPRESS_SCOPE(1);			\
-			mono_array_setref_fast (MONO_HANDLE_RAW (MONO_HANDLE_UNSUPPRESS (HANDLE)), __idx, MONO_HANDLE_RAW (__val)); \
-		} while (0);						\
+		MONO_HANDLE_SUPPRESS_SCOPE(1);			\
```

Comment:
Sorry, this needs to go back to the way it was - the way `MONO_HANDLE_SUPPRESS_SCOPE` is detected by Centrinel is a gross hack and it really has to be the first thing in a `{ compound statement }`

Reply:
Alternately, I think wrapping the `mono_array_setref_fast` call in a `MONO_HANDLE_SUPPRESS()` expression would also work.

Reply:
er, suppress vs. unsupress, nevermind that.